src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,103 @@
+     *
+     * @param input the input string of the user
+     * @return a command object
+     * @throws DukeException when input is not valid
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] words = input.split(" ");
+        String[] stamps = input.split("/");
+        String keyword = words[0];
+        if(input.equals("")){
+            return new ErrorCommand();
+        }
+        if (keyword.equalsIgnoreCase("find")) {
+            if (words.length < 2) {
+                return new ErrorCommand();
+            }
+            String searchingItem = input.substring(5).trim();
+            return new FindCommand(searchingItem);
+        }
+        if (keyword.equalsIgnoreCase("bye")) {
+            return new ExitCommand();
+        }
+        if (input.equalsIgnoreCase("list")) {
+            return new DisplayCommand();
+        }
+        int thisIndex;
+        if (keyword.equalsIgnoreCase("done")) {
+            try {
+                if (words.length != 2) {
+                    throw new DukeException(Message.MESSAGE_INVALIDCOMMAND);
+                }
+                thisIndex = Integer.parseInt(input.substring(5));
+            } catch (Exception exp) {
+                return new ErrorCommand();
+            }
+            return new DoneCommand(thisIndex);
+        } else if (keyword.equalsIgnoreCase("delete")) {
+            try {
+                if (words.length != 2) {
+                    throw new DukeException(Message.MESSAGE_INVALIDCOMMAND);
+                }
+                thisIndex = Integer.parseInt(input.substring(7));
+            } catch (Exception exp) {
+                return new ErrorCommand();
+            }
+            return new DeleteCommand(thisIndex);
+        } else if (keyword.equalsIgnoreCase("todo")) {
+            try {
+                if (words.length < 2) {
+                    throw new DukeException(Message.MESSAGE_INVALIDCOMMAND);
+                }
+            } catch (DukeException exp) {
+                return new ErrorCommand();
+            }
+            return new AddCommand(new Todo(input.substring(5).trim()));
+        } else if (keyword.equalsIgnoreCase("event")) {
+            String[] eventWords = input.split("/at ");
+            try {
+                if (words.length < 4 || stamps.length < 2 || eventWords.length < 2) {
+                    throw new DukeException(Message.MESSAGE_INVALIDCOMMAND);
+                }
+            } catch (DukeException exp) {
+                return new ErrorCommand();
+            }
+            return new AddCommand(new Event(stamps[0].substring(6), eventWords[1]));
+        } else if (keyword.equalsIgnoreCase("deadline")) {
+            String[] ddlWords = input.split("/by ");
+            try {
+                if (words.length < 4 || stamps.length < 2 || ddlWords.length < 2) {
+                    throw new DukeException("The content of a deadline must be complete.");
```

Comment:
Instead of saying "The content of a deadline must be complete.", perhaps you can provide the required format to the user? For eg. Oops! You have to follow the format "deadline [event] /by [date] [time]